src/components/mc/cpu/reduce_alpha/mc_cpu_reduce_alpha.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_MC_CPU_REDUCE_ALPHA_H_
+#define UCC_MC_CPU_REDUCE_ALPHA_H_
+
+#include "components/mc/cpu/reduce/mc_cpu_reduce.h"
+
+#undef OP_N
+#define OP_N(_d, _s1, _s2, _sc, _REDUCE_OP, _n)                                \
```

Comment:
@lappazos do your tests pass when UCC_TL_UCP_RADIX>8 ? Lets try it. I think we have a problem in the implementation there. Your OP_N adds vector op. At the same time you are using DO_DT_REDUCE_FLOAT macro which may call OP_N several times for large n_vectors: e.g. https://github.com/openucx/ucc/blob/b3fb70591fea4085b1b3d14bb12de59fceabf73f/src/components/mc/cpu/reduce/mc_cpu_reduce.h#L63

This means division will be applied multiple times result in the incorrect result. Need to check/resolve. Potentially will need to redefine DO_DT_REDUCE_WITH_OP instead of OP_N and add the vector_op loop in the end there.

Reply:
Fixed